src/child/containers/sidebars/favourites/Favourites.js

```diff
@@ -121,36 +94,93 @@ class Favourites extends Component {
         e.stopPropagation();
     }
 
-    onIconClick(stockCode) {
-        this.setState({ unfavouritingStockCode: stockCode });
-    }
+    onDragStart(e) {
+        const { favourites, bindings } = this.props;
 
-    onDrop(e) {
-        const codes = this.props.favourites.codes;
-        const code = e.dataTransfer.getData('text/plain');
-        this.props.dispatch(insertFavouriteAt(codes.length, code));
-        e.currentTarget.classList.remove('dragOver');
+        this.setState({
+            dragOverIndex: favourites.codes.indexOf(bindings.getCodeFromDT(e.dataTransfer.types)),
+            favouriteHeight: e.target.getBoundingClientRect().height,
+            dragStartClientY: e.nativeEvent.clientY
+        });
     }
 
     onDragOver(e) {
-        if (!e.currentTarget.classList.contains('dragOver')) {
-            e.currentTarget.classList.add('dragOver');
+        const { removeDragOver, props, state } = this;
+        const { favourites, bindings } = props;
+        const { dragOverIndex, favouriteHeight, dragStartClientY } = state;
+        const favouriteElements = this.sidetab.childNodes;
```

Comment:
Again, similar comment to above - I'm really not keen on a parent element selecting elements in a child component and adding/removing classes imperatively, outside of React. If a class needs to be toggled on a child component, then could a prop be passed down to that child component, where it can set the class?